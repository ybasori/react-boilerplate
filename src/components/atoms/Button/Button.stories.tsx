import type { Meta, StoryObj } from "@storybook/react";
import { fn } from "@storybook/test";
import Button from "./Button";

//ðŸ‘‡ This default export determines where your story goes in the story list
const meta: Meta<typeof Button> = {
  title: "Atoms/Button",
  component: Button,
  args: { onClick: fn() },
  argTypes: {
    color: {
      options: ["red", "blue"],
      control: {
        type: "select",
      },
    },
  },
};

export default meta;
type Story = StoryObj<typeof Button>;

export const DefaultButton: Story = {
  args: {
    children: "button",
  },
};
export const BlueButton: Story = {
  args: {
    children: "button",
    color: "blue",
  },
};

export const RedButton: Story = {
  args: {
    children: "button",
    color: "red",
  },
};
