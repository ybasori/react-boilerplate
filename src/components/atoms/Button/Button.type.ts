export interface IButton {
  children: React.ReactNode;
  color?: "blue" | "red";
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
}
