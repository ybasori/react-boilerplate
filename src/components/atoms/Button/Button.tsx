import styles from "./Button.module.scss";
import { IButton } from "./Button.type";

const Button: React.FC<IButton> = ({ color, children, onClick }) => {
  return (
    <button
      className={`${styles["container"]} ${styles["color-" + color]}`}
      onClick={onClick}
    >
      {children}
    </button>
  );
};

export default Button;
