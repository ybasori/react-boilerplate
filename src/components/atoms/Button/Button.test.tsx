import { render, screen, fireEvent } from "@testing-library/react";
import Button from "./Button";
import { sampleData } from "./Button.constant";

describe("Atoms/Button", () => {
  test("renders the button with the correct label", () => {
    render(
      <Button onClick={() => null} color="blue">
        Click
      </Button>
    );

    // Assert that the button is in the document and has the correct text
    const buttonElement = screen.getByText(`${sampleData} Click`);
    expect(buttonElement).toBeInTheDocument();
  });

  test("calls the onClick handler when the button is clicked", () => {
    const handleClick = jest.fn();
    render(
      <Button onClick={handleClick} color="blue">
        Click
      </Button>
    );

    const buttonElement = screen.getByText(`${sampleData} Click`);
    fireEvent.click(buttonElement);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  test("Snapshot", () => {
    const handleClick = jest.fn();
    const { container } = render(
      <Button onClick={handleClick} color="blue">
        Click
      </Button>
    );

    expect(container).toMatchSnapshot();
  });
});
