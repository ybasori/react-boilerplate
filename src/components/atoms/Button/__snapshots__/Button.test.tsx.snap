// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Button Snapshot 1`] = `
<div>
  <button
    class="container color-blue"
  >
    Click
  </button>
</div>
`;
